lib/src/realm_object.dart

```diff
@@ -206,6 +206,16 @@ class RealmObject {
     RealmAccessor.setDefaults<T>(values);
     return true;
   }
+
+  @override
+  // ignore: hash_and_equals
+  bool operator ==(Object other) {
+    final o = other;
```

Comment:
perhaps we don't need an additional local variable.

Reply:
Fixed